cpp/src/arrow/compute/kernels/scalar_validity.cc

```diff
@@ -132,6 +156,11 @@ const FunctionDoc is_null_doc("Return true if null",
                               ("For each input value, emit true iff the value is null."),
                               {"values"});
 
+const FunctionDoc is_nan_doc("Return true if NaN",
+                             ("For each input value, emit true iff the value is NaN "
+                              "(according to std::isnan(value))."),
```

Comment:
I don't think "according to" is necessary, NaN values are standardized by IEEE floating point.

Reply:
Ok